src/syrupy/location.py

```diff
@@ -1,7 +1,29 @@
-from collections import namedtuple
+import inspect
+from typing import (
+    Any,
+    Optional,
+)
 
 
-TestLocation = namedtuple(
-    "TestLocation",
-    ["filename", "modulename", "classname", "methodname", "nodename", "testname"],
-)
+class TestLocation(object):
+    def __init__(self, node: Any):
+        self._node = node
+        self.filename = self._node.fspath
+        self.modulename = self._node.obj.__module__
+        self.methodname = self._node.obj.__name__
+        self.nodename = getattr(self._node, "name", None)
+        self.testname = self.nodename or self.methodname
+
+    @property
+    def classname(self) -> Optional[str]:
```

Comment:
Not sure if there's a better way to extract the class name from the `request.node` when used in `session.py#196` the `node` does have a `request` object attached